hevc/pps.go

```diff
@@ -0,0 +1,624 @@
+				ext.ChromaBitDepthEntryMinus8 = r.ReadExpGolomb()
+			}
+			ext.PalettePredictorInitializer = make([][]uint, numComps)
+			// Fill luma
+			for i := uint(0); i < ext.NumPalettePredictorInitializers; i++ {
+				ext.PalettePredictorInitializer[0] =
+					append(ext.PalettePredictorInitializer[0], r.Read(int(ext.LumaBitDepthEntryMinus8+8)))
+			}
+			// Fill chroma if any
+			for comp := 1; comp < numComps; comp++ {
+				for i := uint(0); i < ext.NumPalettePredictorInitializers; i++ {
+					ext.PalettePredictorInitializer[comp] =
+						append(ext.PalettePredictorInitializer[comp], r.Read(int(ext.ChromaBitDepthEntryMinus8+8)))
+				}
+			}
+		}
+	}
+
+	if r.AccError() != nil {
+		return nil, r.AccError()
+	}
+
+	return ext, nil
+}
+
+func parse3dExtension(r *bits.AccErrEBSPReader) (*D3Extension, error) {
+	ext := &D3Extension{}
+	ext.DltsPresentFlag = r.ReadFlag()
+	if ext.DltsPresentFlag {
+		ext.NumDepthLayersMinus1 = uint8(r.Read(6))
+		ext.BitDepthForDepthLayersMinus8 = uint8(r.Read(4))
+		for i := uint8(0); i <= ext.NumDepthLayersMinus1; i++ {
+			layer := DepthLayer{}
+			layer.DltFlag = r.ReadFlag()
+			if layer.DltFlag {
+				layer.DltPredFlag = r.ReadFlag()
+				if !layer.DltPredFlag {
+					layer.DltValFlagsPresentFlag = r.ReadFlag()
+				}
+				if layer.DltValFlagsPresentFlag {
+					// variable depthMaxValue is set equal to ( 1 << ( pps_bit_depth_for_depth_layers_minus8 + 8 ) ) − 1
+					depthMaxValue := (1 << (ext.BitDepthForDepthLayersMinus8 + 8)) - 1
+					for j := 0; j <= depthMaxValue; j++ {
+						layer.DltValueFlag = append(layer.DltValueFlag, r.ReadFlag())
+					}
+				} else {
+					var err error
+					layer.DeltaDlt, err = parseDeltaDlt(r, int(ext.BitDepthForDepthLayersMinus8+8))
+					if err != nil {
+						return ext, err
+					}
+				}
+			}
+			ext.DepthLayers = append(ext.DepthLayers, layer)
+		}
+	}
+
+	if r.AccError() != nil {
+		return nil, r.AccError()
+	}
+
+	return ext, nil
+}
+
+func parseDeltaDlt(r *bits.AccErrEBSPReader, BitDepthForDepthLayers int) (*DeltaDlt, error) {
+	dd := &DeltaDlt{}
+	dd.NumValDeltaDlt = r.Read(BitDepthForDepthLayers)
+	if dd.NumValDeltaDlt > 0 {
+		if dd.NumValDeltaDlt > 1 {
+			dd.MaxDiff = r.Read(BitDepthForDepthLayers)
+		}
+		if dd.NumValDeltaDlt > 2 && dd.MaxDiff > 0 {
+			dd.MinDiffMinus1 = r.Read(ceilLog2(dd.MaxDiff + 1))
+		} else {
+			dd.MinDiffMinus1 = dd.MaxDiff - 1
+		}
+		dd.DeltaDltVal0 = r.Read(BitDepthForDepthLayers)
+		if dd.MaxDiff > (dd.MinDiffMinus1 + 1) {
+			for k := uint(1); k < dd.NumValDeltaDlt; k++ {
+				// variable minDiff is set equal to ( min_diff_minus1 + 1 )
+				// length of delta_val_diff_minus_min[ k ] syntax element is Ceil( Log2( max_diff − minDiff + 1 ) ) bits
+				dd.DeltaValDiffMinusMin =
+					append(dd.DeltaValDiffMinusMin, r.Read(ceilLog2(dd.MaxDiff-(dd.MinDiffMinus1+1)+1)))
+			}
+		}
+	}
+
+	if r.AccError() != nil {
+		return nil, r.AccError()
+	}
+
+	return dd, nil
+}
+
+// ceilLog2 - nr bits needed to represent numbers 0 - n-1 values
```

Comment:
This is now a function exported from the bits package, so please use the exported function.